nifi-api/src/main/java/org/apache/nifi/controller/status/PerformanceMetrics.java

```diff
@@ -0,0 +1,234 @@
+
+    public String getIdentifier() {
+        return identifier;
+    }
+
+    public void setIdentifier(String identifier) {
+        this.identifier = identifier;
+    }
+
+    public long getCpuTime() {
+        return cpuTime;
+    }
+
+    public void setCpuTime(long cpuTime) {
+        this.cpuTime = cpuTime;
+    }
+
+    public long getCpuTimePercentage() {
+        return cpuTimePercentage;
+    }
+
+    public void setCpuTimePercentage(long cpuTimePercentage) {
+        this.cpuTimePercentage = cpuTimePercentage;
+    }
+
+    public long getReadTime() {
+        return readTime;
+    }
+
+    public void setReadTime(long readTime) {
+        this.readTime = readTime;
+    }
+
+    public long getReadTimePercentage() {
+        return readTimePercentage;
+    }
+
+    public void setReadTimePercentage(long readTimePercentage) {
+        this.readTimePercentage = readTimePercentage;
+    }
+
+    public long getWriteTime() {
+        return writeTime;
+    }
+
+    public void setWriteTime(long writeTime) {
+        this.writeTime = writeTime;
+    }
+
+    public long getWriteTimePercentage() {
+        return writeTimePercentage;
+    }
+
+    public void setWriteTimePercentage(long writeTimePercentage) {
+        this.writeTimePercentage = writeTimePercentage;
+    }
+
+    public long getCommitTime() {
+        return commitTime;
+    }
+
+    public void setCommitTime(long commitTime) {
+        this.commitTime = commitTime;
+    }
+
+    public long getCommitTimePercentage() {
+        return commitTimePercentage;
+    }
+
+    public void setCommitTimePercentage(long commitTimePercentage) {
+        this.commitTimePercentage = commitTimePercentage;
+    }
+
+    public long getGcTime() {
+        return gcTime;
+    }
+
+    public void setGcTime(long gcTime) {
+        this.gcTime = gcTime;
+    }
+
+    public long getGcTimePercentage() {
+        return gcTimePercentage;
+    }
+
+    public void setGcTimePercentage(long gcTimePercentage) {
+        this.gcTimePercentage = gcTimePercentage;
+    }
+
+    public long getBytesRead() {
+        return bytesRead;
+    }
+
+    public void setBytesRead(long bytesRead) {
+        this.bytesRead = bytesRead;
+    }
+
+    public long getBytesReadPercentage() {
+        return bytesReadPercentage;
+    }
+
+    public void setBytesReadPercentage(long bytesReadPercentage) {
+        this.bytesReadPercentage = bytesReadPercentage;
+    }
+
+    public long getBytesWritten() {
+        return bytesWritten;
+    }
+
+    public void setBytesWritten(long bytesWritten) {
+        this.bytesWritten = bytesWritten;
+    }
+
+    public long getBytesWrittenPercentage() {
+        return bytesWrittenPercentage;
+    }
+
+    public void setBytesWrittenPercentage(long bytesWrittenPercentage) {
+        this.bytesWrittenPercentage = bytesWrittenPercentage;
+    }
+
+    public static void merge(final PerformanceMetrics target, final PerformanceMetrics toMerge) {
```

Comment:
This should not be implemented in the class. Instead, it should be implemented in a separate `Merger` implementation inside the framework.